packages/mui-material/src/Slider/Slider.js

```diff
@@ -457,25 +459,26 @@ const Slider = React.forwardRef(function Slider(inputProps, ref) {
         ...componentsProps,
         root: {
           ...componentsProps.root,
-          ...(shouldSpreadOwnerState(components.Root) && {
+          ...(shouldSpreadAdditionalProps(components.Root) && {
+            as: component,
```

Comment:
This was previously handled in the `SliderUnstyled`. The logic was moved here now.